File: src/main/java/io/junseok/mealmate/domain/memberwish/presentation/WishController.java
package io.junseok.mealmate.domain.memberwish.presentation;

import io.junseok.mealmate.domain.memberwish.dto.response.MemberWishList;
import io.junseok.mealmate.domain.memberwish.service.WishService;
import java.security.Principal;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/wish")
@RequiredArgsConstructor
@CrossOrigin
public class WishController {
    private final WishService wishService;

    @GetMapping("/{restaurantId}")
    public ResponseEntity<Long> saveWishList(@PathVariable Long restaurantId, Principal principal){
        return ResponseEntity.ok(wishService.createWishList(restaurantId,principal.getName()));
    }

    @GetMapping
    public ResponseEntity<MemberWishList> getMemberWishLists(Principal principal){
        return ResponseEntity.ok(wishService.getWishList(principal.getName()));
    }

    @DeleteMapping("/{wishListId}")
    public ResponseEntity<Void> deleteWishList(@PathVariable Long wishListId){
        wishService.remove(wishListId);
        return ResponseEntity.ok().build();
    }

    @GetMapping("/count")
    public ResponseEntity<Integer> getWishListCount(Principal principal){
        return ResponseEntity.ok(wishService.showWishListCount(principal.getName()));
    }
}

File: src/main/java/io/junseok/mealmate/domain/memberwish/presentation/WishController.kt
package io.junseok.mealmate.domain.memberwish.presentation

import io.junseok.mealmate.domain.memberwish.dto.response.MemberWishList
import io.junseok.mealmate.domain.memberwish.service.WishService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.security.Principal

@RestController
@RequestMapping("/api/wish")
@CrossOrigin
class WishController(
    private val wishService: WishService
) {

    @GetMapping("/{restaurantId}")
    fun saveWishList(
        @PathVariable restaurantId: Long,
        principal: Principal
    ): ResponseEntity<Long> =
        ResponseEntity.ok(wishService.createWishList(restaurantId, principal.name))


    @GetMapping
    fun getMemberWishLists(principal: Principal): ResponseEntity<MemberWishList> =
        ResponseEntity.ok(wishService.getWishList(principal.name))

    @DeleteMapping("/{restaurantId}")
    fun deleteWishList(
        @PathVariable restaurantId: Long,
        principal: Principal
    ): ResponseEntity<Unit> {
        wishService.remove(restaurantId, principal.name)
        return ResponseEntity.ok().build()
    }

    @GetMapping("/count")
    fun getWishListCount(principal: Principal): ResponseEntity<Int> =
        ResponseEntity.ok(wishService.showWishListCount(principal.name))
}
